make-base/build.mk

```diff
@@ -0,0 +1,19 @@
+.PHONY: always
+
+define make_build_targets
+$(1)/build/image: $(1)/build/latest-change
+	mkdir -p $(1)/build
+
+$(1)/build/latest-change: always
+	mkdir -p $(1)/build
+
+.PHONY: $(1)/clean
+$(1)/clean:
+	rm -rf $(1)/build
+endef
+
+$(eval $(call make_build_targets,\
```

Comment:
This logic here will ensure that regardless of where projects are organized (top level or inside of sub directory), they have a fixed reference.